### PR TITLE
[web] Migrate Flutter Web DOM usage to JS static interop - 42.

### PR DESCRIPTION
**Pull request by joshualitt**
This is CL 42 in a series of CLs to migrate Flutter Web DOM usage to the new JS static interop API.

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 89 new digest(s) on patchset 33.
View them at https://flutter-engine-gold.skia.org/cl/github/33378

File: lib/web_ui/lib/src/engine/dom.dart [hunk 1]
```diff
@@ -682,18 +682,21 @@ DomXMLHttpRequest createDomXMLHttpRequest() =>
...
   void open(String method, String url, [bool? async]) => js_util.callMethod(
       this, 'open', <Object>[method, url, if (async != null) async]);
-  external void send();
+  void send([Object? bodyOrData]) => js_util
+      .callMethod(this, 'send', <Object>[if (bodyOrData != null) bodyOrData]);
 }
 
-Future<DomXMLHttpRequest> domHttpRequest(String url, {String? responseType}) {
+Future<DomXMLHttpRequest> domHttpRequest(String url,
+    {String? responseType, String? method, dynamic sendData}) {
```
Comment (eyebrowsoffire) [inline lib/web_ui/lib/src/engine/dom.dart:696]: `String method = 'GET'`?

File: lib/web_ui/lib/src/engine/dom.dart [hunk 2]
```diff
@@ -713,7 +716,11 @@ Future<DomXMLHttpRequest> domHttpRequest(String url, {String? responseType}) {
   }));
 
   xhr.addEventListener('error', allowInterop(completer.completeError));
-  xhr.send();
+  if (sendData != null) {
```
Comment (eyebrowsoffire) [inline lib/web_ui/lib/src/engine/dom.dart:719]: Do you really need this null check here? There is one inside the method itself. `xhr.send()` is equivalent to `xhr.send(null)` anyway, right?

**Review by eyebrowsoffire — APPROVED** [review]
Small nitpicks, otherwise looks good!

**Pull request closed without merging**